entity/validate_test.go

```diff
@@ -128,25 +87,29 @@ func TestValidateWriteOpOK(t *testing.T) {
 		Caller:     "dn",
 	}
 	err := validate.WriteOp(wo)
-	if err == nil {
-		t.Fatal("err is nill, expected an enitty.ValidationError")
-	}
-	v, ok := err.(entity.ValidationError)
-	if !ok {
-		t.Fatalf("err is type %#v, expected entity.ValidationError", err)
-	}
-	if v.Type != "invalid-entity-type" {
-		t.Errorf("Type = %s, expected invalid-entity-type", v.Type)
-	}
+	require.Error(t, err)
+	assertValidationError(t, err, "invalid-entity-type")
 }
 
 func TestValidateDeleteLabel(t *testing.T) {
 	err := validate.DeleteLabel("foo")
-	if err != nil {
-		t.Errorf("got err '%v', expected nil", err)
-	}
+	require.NoError(t, err)
 	err = validate.DeleteLabel("_id")
+	require.Error(t, err)
+}
+
+// assertValidationError asserts the error to be a non-nil ValidationError and asserts the expected type.
+func assertValidationError(t *testing.T, err error, expectedType string) {
+	// Ugly asserts and returns instead of require so that the test can continue
+	t.Helper()
+	assert.Error(t, err)
```

Comment:
Not sure why `assert.Error` is needed here since the function is used to verify the error type and the verification of whether an err exists is part of the usage pattern. Seems like you can just check if the err is nil and return if true.

Reply:
Per the method comments `assertValidationError asserts the error to be a non-nil ValidationError and asserts the expected type.` This is used to check that:
1. The error is non-nill (and fail the test if it is nil)
2. The error is a validation error (and fail the test if it is not)
3. The Type matches the expectedType (and fail the test if it is not)